python/vineyard/core/client.py

```diff
@@ -622,27 +635,33 @@ def _find_the_most_available_memory_instance(self) -> int:
             int: The instance id with the most available memory.
             if only have one instance, return -1
         """
+        os.environ['VINEYARD_RPC_SKIP_RETRY'] = '1'
```

Comment:
Use `with vineyard.envvars()....` to avoid the global variable been populated.

Reply:
Make sense. Thanks.